File: heranca/cpp/Funcionario.cpp
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Herança em C++
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

#include "Funcionario.hpp"
#include <iostream>

using namespace std;

Funcionario::Funcionario()
{
	cout << "\nExecutando construtor da classe Funcionario";
}

float Funcionario::getSalario()
{
    return salario;
}

void Funcionario::setSalario(float s)
{
    salario = s;
}

string Funcionario::getCargo()
{
    return cargo;
}

void Funcionario::setCargo(string c)
{
    cargo = c;
}

File: heranca/java/Funcionario.java
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Herança em Java
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

public class Funcionario extends Pessoa {
	private float salario;
	private String cargo;

	public Funcionario() {
		System.out.println("Executando construtor da classe Funcionario");
	}

	public float getSalario() {
		return salario;
	}

	public void setSalario(float salario) {
		this.salario = salario;
	}

	public String getCargo() {
		return cargo;
	}

	public void setCargo(String cargo) {
		this.cargo = cargo;
	}
}